packages/core/src/components/Form.js

```diff
@@ -6,18 +6,23 @@ import _isEmpty from "lodash/isEmpty";
 
 import { default as DefaultErrorList } from "./ErrorList";
 import {
-  getDefaultFormState,
   retrieveSchema,
   shouldRender,
-  toIdSchema,
-  getDefaultRegistry,
   deepEquals,
   toPathSchema,
   isObject,
+  getRegistry,
+  validate,
+  getStateFromProps,
 } from "../utils";
-import validateFormData, { toErrorList } from "../validate";
+import { toErrorList } from "../validate";
 import { mergeObjects } from "../utils";
 
+function handleChange(props, state) {
+  const { lastProps, ...formState } = state;
```

Comment:
On [line 62](https://github.com/rjsf-team/react-jsonschema-form/pull/2010/files#diff-a3830d0e95feb07e61c8e35cf02715c655766e69871111b67373658e5d98398bR62) we need to compare prop changes to determine if the consumer has updated their props.

If the consumer has updated the props, we update the internal state based on the props.

We could previously do this in componentWillReceiveProps since that method supports both viewing previous and new props. But since getDerivedStateFromProps only shows the new prop updates, we need to store the old props in the state.

Reply:
And the reason why we cannot always derive state from props, is due to us supporting the mixed controlled/uncontrolled behavior we discussed on our last meeting this Friday.